tests/core/test_device_collector.py

```diff
@@ -56,19 +61,69 @@ def test_device_connector_sync_run_engine_created_connects(RE):
     assert working_device.connected
 
 
+def test_device_connector_async_run_engine_same_event_loop():
+    async def set_up_device():
+        async with DeviceCollector(sim=True):
+            sim_motor = motor.Motor("BLxxI-MO-TABLE-01:X")
+        return sim_motor
+
+    loop = asyncio.new_event_loop()
+    checking_loop = asyncio.new_event_loop()
+
+    try:
+        sim_motor = loop.run_until_complete(set_up_device())
+        RE = RunEngine(call_returns_result=True, loop=loop)
+
+        def my_plan():
+            yield from bps.mov(sim_motor, 3.14)
+
+        RE(my_plan())
+
+        assert (
+            checking_loop.run_until_complete(sim_motor.setpoint.read())[
+                "sim_motor-setpoint"
+            ]["value"]
+            == 3.14
+        )
+
+    finally:
+        if RE.state not in ("idle", "panicked"):
+            try:
+                RE.halt()
+            except TransitionError:
+                pass
+        loop.call_soon_threadsafe(loop.stop)
+        checking_loop.call_soon_threadsafe(checking_loop.stop)
```

Comment:
Could extract both loop and RE out into fixtures and handle the teardown logic there. Not essential but would make the test clearer.

Reply:
I thought about it, but to represent the RE-independent connection of the `DeviceConnector` we'd either have to:

* Initialise the `RunEngine` with the `new_event_loop` in the fixture, and the `DeviceConnector` in the test with the same loop. I don't like this much because we want to go through the device connection prior to any exisiting run-engine - we want to check for future additions of `call_in_bluesky_event_loop` in the codebase, but also because that will be how it'll be done in real devices.
* Initialise the `DeviceCollector` in the fixture before run-engine initialisation with the same event-loop - which would lead to a fixture as complicated as the test, and only being used for one test.